### ISSUE TITLE
Suggestion of minor changes

### ISSUE THREAD

**Issue by rfmsimoes** [jakubgarfield/Bonobo-Git-Server#641]
hello everyone,

I'm new using this git server.
It seems something light and powerful.

However I would like to know your opinion about some topics:

1 - Why user (without admin rights) when look into the list os users belong to a team and press the "back button" is redirect into the list of all team (Teams/Index) intead to be redirect to the previous page?

If tou try to redirect someone to the lnedx of Team without the admin right a error about permission will show...I think it will be better to use something like: href="@Request.UrlReferrer" instead of   href="@Url.Action("Index")

2 - In repository details instead showing part of the URL (when I was exploring the server I noticed that) like http://localhost(....) why not show the machine name?

In SetGitUrls function in Repository Contrroller instead using the Request.Url.Host why not using something like System.Net.Dns.GetHostName() ?

Because I can access a server by name or by IP...but if I try to send the link to someone the PC name will be more easy and the advantage of this approach is even I connected directly in the machine by remote desktop I can copy past the url in repository detail without care about the localhost part.

3 - Why a user (even without admin rights) can change their username? shouldn't it be readonly?
shouldn't be only admin or users with admin rights be able to do that?

I know this is nothing very important and with a low impact in everything.
But I just would like to know your opinion. ( I've done this changes at my side =D )

And by the way, what do you think about the possibility to be able to add more roles than administrator? I know this could have a hudge impact in all website...but just one idea. :)

**Comment by willdean** [comment]
@rfmsimoes - Thanks for moving this - it would still be better as one issue per topic, but don't worry for now.

1.  I can't repro this - could you provide a URL for the page you're on where the back button doesn't work properly?

3. I guess we think that a user's name is something he can control.  In 6.0 the user's name is not his identity (that's a guid), so the name is just cosmetic really.

Re adding more roles, the underlying data storage could cope with this, but I think you might struggle to get agreement about what the roles would be.  All authorisation systems are either too complicated or not complicated enough.

**Comment by rfmsimoes** [comment]
@willdean  thanks for your reply :)

I understand your point of view about the roles.
You are right that clearly will create a hudge complexity and some disagreement  =)
For a light and simple server maybe it is best not complicate to much 


About the other items.
I will try to explain a little bit better.
But again, it is just one personal opinion and nothing else.  =)

Item1:
![image](https://cloud.githubusercontent.com/assets/21081206/22515687/3b139348-e89b-11e6-8739-2623fe221675.png)


Item 3:
![image](https://cloud.githubusercontent.com/assets/21081206/22515524/a4f5d1f0-e89a-11e6-9cad-c8a0cf6e1b23.png)

**Comment by larshg** [comment]
@rfmsimoes I just tested it and it checks if a username is valid/used or not - so its only cosmetic and there can only be a one unique username in the DB.

**Comment by rfmsimoes** [comment]
@larshg 

I have a different point of view about the username....but it is ok for me if you thought on that as a cosmetic thing. =)

About the back button ... Glad we are in agreement =)

**Comment by willdean** [comment]
I don't think we should provide a 'back' button at all - that's the browser's job.
